### PR TITLE
Implement anyhow context: dependency_injection

### PR DESCRIPTION
**Pull request by dlachaumepalo**
## Content
This PR includes an update of mithril-aggregator `dependency_injection` with `anyhow` context.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [X] Commit sequence broadly makes sense
  - [X] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [X] Self-reviewed the diff
  - [X] Useful pull request description
  - [x] Reviewer requested
- Documentation
  - [ ] Update README file (if relevant)
  - [ ] Update documentation website (if relevant)
  - [ ] Add dev blog post (if relevant)

## Issue(s)
Relates to #798

### CODE REVIEW

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

File: mithril-aggregator/src/commands/genesis_command.rs [hunk 1]
```diff
@@ -73,7 +73,10 @@ impl ExportGenesisSubCommand {
             self.target_path.display()
         );
         let mut dependencies_builder = DependenciesBuilder::new(config.clone());
-        let dependencies = dependencies_builder.create_genesis_container().await?;
+        let dependencies = dependencies_builder
+            .create_genesis_container()
+            .await
+            .with_context(|| "Dependencies Builder can not create genesis container")?;
```
Comment (ghubertpalo) [inline mithril-aggregator/src/commands/genesis_command.rs:79]: `genesis container` is vague when out of context. `cannot create genesis command dependencies container` would imho suit better.

File: mithril-aggregator/src/commands/genesis_command.rs [hunk 2]
```diff
@@ -105,7 +108,10 @@ impl ImportGenesisSubCommand {
             self.signed_payload_path.to_string_lossy()
         );
         let mut dependencies_builder = DependenciesBuilder::new(config.clone());
-        let dependencies = dependencies_builder.create_genesis_container().await?;
+        let dependencies = dependencies_builder
+            .create_genesis_container()
+            .await
+            .with_context(|| "Dependencies Builder can not create genesis container")?;
```
Comment (ghubertpalo) [inline mithril-aggregator/src/commands/genesis_command.rs:114]: same as above

File: mithril-aggregator/src/commands/genesis_command.rs [hunk 3]
```diff
@@ -170,7 +176,10 @@ impl BootstrapGenesisSubCommand {
         debug!("BOOTSTRAP GENESIS command"; "config" => format!("{config:?}"));
         println!("Genesis bootstrap for test only!");
         let mut dependencies_builder = DependenciesBuilder::new(config.clone());
-        let dependencies = dependencies_builder.create_genesis_container().await?;
+        let dependencies = dependencies_builder
+            .create_genesis_container()
+            .await
+            .with_context(|| "Dependencies Builder can not create genesis container")?;
```
Comment (ghubertpalo) [inline mithril-aggregator/src/commands/genesis_command.rs:182]: same as above

**Review by ghubertpalo — APPROVED** [review]

**Pull request merged**